tracer/src/Datadog.Trace/Propagators/SpanContextPropagator.cs

```diff
@@ -38,45 +35,6 @@ internal SpanContextPropagator(
             _extractors = extractors?.ToArray() ?? [];
         }
 
-        public static SpanContextPropagator Instance
```

Comment:
This is causing a lot of code changes and a less discoverable API ("How do I get the `SpanContextPropagator`?")
What do you think of leaving `SpanContextPropagator.Instance` as a getter-only helper:

```csharp
public static SpanContextPropagator Instance => Trace.Instance.TracerManager.SpanContextPropagator;
```

Reply:
I'd rather have the dependency on the tracer be explicit. When you have a local instance of the tracer you should use it instead of Tracer.Instance, having SpanContextPropagator.Instance encourages to do the wrong thing.

I believe @andrewlock would agree (we had a somewhat similar discussion before)

Reply:
> When you have a local instance of the tracer you should use it instead of `Tracer.Instance`

Yeah, agreed.

And adding a `Tracer.Instance.SpanContextPropagator => Tracer.Instance.TracerManager.SpanContextPropagator` helper isn't that helpful. Thanks.